modules/native/native-grammar.ym

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2025 BalaBit
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+%code top {
+#include "native-parser.h"
+
+}
+
+
+%code {
+
+#include "cfg-parser.h"
+#include "parser.h"
+#include "native-grammar.h"
+#include "syslog-names.h"
+#include "messages.h"
+#include "plugin.h"
+#include "cfg-grammar.h"
+
+#include <string.h>
+
+__attribute__((visibility("hidden"))) LogParser*
+native_parser_new(GlobalConfig *cfg);
+
+}
+
+%name-prefix "native_"
+
+/* this parameter is needed in order to instruct bison to use a complete
+ * argument list for yylex/yyerror */
+
+%lex-param {CfgLexer *lexer}
+%parse-param {CfgLexer *lexer}
+%parse-param {LogParser **instance}
+%parse-param {gpointer arg}
+
+/* INCLUDE_DECLS */
+
+%token KW_OPTION
+
+%type <ptr> native
+%type <ptr> native_parser
+%type <ptr> native_parser_params
+%%
+
+start
+  : native
+    {
+      *instance = $1;
+      if (yychar != YYEMPTY)
+        cfg_lexer_unput_token(lexer, &yylval);
+      YYACCEPT;
+    }
+  ;
+
+native
+  : LL_CONTEXT_PARSER native_parser { $$ = $2; }
+  ;
+
+native_parser
+  : LL_IDENTIFIER '(' native_parser_params ')' { $$ = *instance; }
+  ;
+
+native_parser_params
+  : {
+      *instance = native_parser_new(configuration);
+
+      if (!*instance)
+       {
+         msg_error("Cannot create the Native parser, aborting",
+                   NULL);
+         YYERROR;
+       }
+    }
+    native_parser_options
+  ;
+
+native_parser_options
+  : native_parser_option native_parser_options
+  |
+  ;
+
+native_parser_option
+  : KW_OPTION '(' string string ')'
+    {
+      native_parser_set_option(*instance, $3, $4);
```

Comment:
You just call the lex function which returns tokens YYSTYPE). If you can call the lexer implemented in C you will be wired into the grammar system.

In fact this could work with all the other language bindings we have so far.

Also it'd be nice if the proxy would be simpler, can't you fill the LogParser struct directly from rust?